src/components/HardwareAccountSelect.js

```diff
@@ -30,7 +30,7 @@ function HardwareAccountSelect({ type, path, onClose, confirmAddress }) {
       confirmAddress(address);
       onClose();
     }, onClose);
-  }, [confirmAddress, connect, onClose]);
+  }, []);
```

Comment:
hmm on first glance, i thought the dep array was ok here -- was it causing trouble as it was?

Reply:
Passing `connect` was causing the problem of not loading ledger addresses in the UI. After the addresses were fetched, a new instance was created which overwrote the addresses in state back to `initialState` default of `[]`. I added `confirmAddress` and `onClose` back to the deps array because they didn't have any effect.

Reply:
The root cause is that when `connect` is defined inside `useHardwareWallet`, it's not using `useCallback`. https://reactjs.org/docs/hooks-reference.html#usecallback